src/controllers/auth-controller.ts

```diff
@@ -99,6 +101,25 @@ export default class AuthController extends AbstractController {
     }
   };
 
+  private changePassword = async (req: Request, res: Response) => {
+    const { oldPassword, newPassword } = req.body;
+    const user = req.user as User;
+    if (!oldPassword || !newPassword || !user) {
+      return res.status(400).end();
+    }
+    const isPasswordCorrect = user ? await isCorrect(oldPassword, user.salt, user.hash) : false;
```

Comment:
redundant ternary expression, `user` exists here

Reply:
yep, noted it too, you checked user before

Reply:
yep, noted it too, you checked user before

Reply:
fixed